public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx

```diff
@@ -0,0 +1,257 @@
+
+const containerElementName = 'root';
+
+interface State {
+    options: any;
+}
+
+interface Props {
+    genderAnalysis: any;
+    genderCountAnalysis: any;
+    selectedResult: any;
+    domain: string;
+}
+
+export class BioSexChartReactComponent extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props);
+    this.state = {options: null};
+  }
+
+  componentDidMount() {
+      this.getChartOptions();
+  }
+
+  getChartOptions() {
+    if (this.props.domain === 'fitbit') {
+        this.getFitbitChartOptions();
+    } else {
+        this.getSurveyChartOptions();
+    }
+  }
+
+  getFitbitChartOptions() {
+    baseOptions.chart.type = 'bar';
+    baseOptions.plotOptions.series.pointWidth = 50;
+    baseOptions.yAxis.title.text = 'Participant Count';
+    baseOptions.xAxis.title.text = '';
+    baseOptions.yAxis.title.style.fontSize = '14px';
+    baseOptions.xAxis.title.style.fontSize = '14px';
+    baseOptions.yAxis.title.style.color = '#262262';
+    baseOptions.yAxis.gridLineColor = '#F0F0F0';
+    const {categories, data} = this.prepFitbitCategoriesAndData();
+    baseOptions.xAxis.categories = categories;
+    baseOptions.series[0].data = data;
+    this.setState({options: baseOptions});
+  }
+
+  getSurveyChartOptions() {
+    baseOptions.chart.type = 'column';
+    baseOptions.plotOptions.column.groupPadding = 0.40;
+    baseOptions.plotOptions.series.pointWidth = 50;
+    baseOptions.legend.enabled = true;
+    baseOptions.yAxis.gridLineColor = '#ECF1F4';
+    baseOptions.title.style.color = '#262262';
+    baseOptions.title.style.fontSize = '22px';
+    baseOptions.color = '#2691D0';
+    baseOptions.xAxis.title.text = this.props.genderAnalysis.analysisName;
+    baseOptions.yAxis.title.text = 'Participant Count';
+    this.props.genderAnalysis.results = this.props.genderAnalysis.results.filter(
+              r => r.stratum4 === this.props.selectedResult.stratum4);
+    const {categories, series} = this.prepSurveyCategoriesAndData();
+    baseOptions.xAxis.categories = categories;
+    if ('dataOnlyLT20' in series[0]) {
+        baseOptions.yAxis.min = series[0].dataOnlyLT20 ? 20 : 0;
+        baseOptions.yAxis.labels = {
+            style: {
+                   fontSize: '14px',
+                   whiteSpace: 'wrap',
+                   textOverflow: 'ellipsis',
+                   color: '#262262'
+            },
+            formatter: function () {
+                   const label = this.axis.defaultLabelFormatter.call(this);
+                   // Change <= 20 count to display '<= 20'
+                   if (label <= 20) {
+                    return '&#8804; 20';
+                   }
+                   return label;
+                   },
+                   useHTML: true
+            };
+    }
+    baseOptions.series = series;
+    console.log(baseOptions);
+    this.setState({options: baseOptions});
+  }
+
+  prepSurveyCategoriesAndData() {
```

Comment:
Nit: Pull props here using nested properties again. I also added the arg from the above comment:
```suggestion
  prepSurveyCategoriesAndData(genderAnalysisResults) {
    const {genderAnalysis: {analysisId}, genderCountAnalysis: {results}} = this.props;
```

Reply:
Changed this.